modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ues/ui/integration/test/dashboard/BannerTest.java

```diff
@@ -0,0 +1,384 @@
+
+    private static final String EDITOR_ROLE = "editor";
+    private static final String VIEWER_ROLE = "viewer";
+    private static final String ADMIN_ROLE = "admin";
+
+    private ResourceAdminServiceClient resourceAdminServiceClient;
+    private User editor, viewer;
+
+    /**
+     * Initialize the class
+     */
+    public BannerTest() throws Exception {
+        super(TestUserMode.SUPER_TENANT_ADMIN);
+
+        AutomationContext automationContext = new AutomationContext(UESIntegrationTestConstants
+                .UES_PRODUCT_NAME, this.userMode);
+
+        editor = automationContext.getContextTenant().getTenantUser("editor");
+        viewer = automationContext.getContextTenant().getTenantUser("viewer");
+
+        resourceAdminServiceClient = new ResourceAdminServiceClient(getBackEndUrl(),
+                getCurrentUsername(), getCurrentPassword());
+
+        UserManagementClient userManagementClient = new UserManagementClient(getBackEndUrl(),
+                getCurrentUsername(), getCurrentPassword());
+
+        // Create user roles
+        String[] editorsRoles = new String[]{editor.getUserName()};
+        String[] viewersRoles = new String[]{viewer.getUserName()};
+
+        userManagementClient.addRole(EDITOR_ROLE, editorsRoles, null);
+        userManagementClient.addRole(VIEWER_ROLE, viewersRoles, null);
+
+        // Remove unnecessary roles from users
+        String[] rolesToRemove = new String[]{ADMIN_ROLE};
+
+        userManagementClient.addRemoveRolesOfUser(editor.getUserName(), null, rolesToRemove);
+        userManagementClient.addRemoveRolesOfUser(viewer.getUserName(), null, rolesToRemove);
+    }
+
+    /**
+     * Setup the testing environment
+     *
+     * @throws Exception
+     */
+    @BeforeClass(alwaysRun = true)
+    public void setUp() throws Exception {
+
+        // login as admin and create the dashboard
+        login(getDriver(), getBaseUrl(), getCurrentUsername(), getCurrentPassword());
+
+        initDashboard();
+
+        logout(getDriver(), getBaseUrl(), getCurrentUsername());
+    }
+
+    /**
+     * Cancel adding banner by editor
+     *
+     * @throws Exception
+     */
+    @Test(groups = "wso2.ues.dashboard", description = "Cancel adding banner by editor")
+    public void cancelAddingBannerByEditor() throws Exception {
+
+        login(getDriver(), getBaseUrl(), editor.getUserName(), editor.getPassword());
+
+        goToDesigner();
+
+        clickEditBannerButton(0);
+        clickCancelBannerButton();
+        assertEquals(isResourceExist(DASHBOARD_ID + "/banner"), false,
+                "Resource should not be uploaded to the registry");
+    }
+
+    /**
+     * Add banner by editor
+     *
+     * @throws Exception
+     */
+    @Test(groups = "wso2.ues.dashboard", description = "Add banner by editor",
+            dependsOnMethods = "cancelAddingBannerByEditor")
+    public void addBannerByEditor() throws Exception {
```

Comment:
check the possibility of throwing the specific exception in all test methods. Might require some refactoring of calling method signatures